web/src/composebox_typeahead.ts

```diff
@@ -1029,62 +1282,68 @@ export function content_typeahead_selected(item, query, input_element, event) {
         if (highlight.start && highlight.end) {
             $textbox.range(highlight.start, highlight.end);
         } else {
-            $textbox.caret(beginning.length, beginning.length);
+            // TODO(prep) commit for removing unecessary argument
+            $textbox.caret(beginning.length);
         }
         // Also, trigger autosize to check if compose box needs to be resized.
         compose_ui.autosize_textarea($textbox);
     }, 0);
     return beginning + rest;
 }
 
-export function compose_content_matcher(completing, token) {
-    switch (completing) {
+export function compose_content_matcher(item: PillItem, token: string): boolean {
+    switch (item.type) {
         case "emoji":
-            return typeahead.get_emoji_matcher(token);
+            return typeahead.get_emoji_matcher(token)(item);
```

Comment:
I might be able to do a prep commit for changing this function into something that _does_ the matching instead of returning a matcher. I should maybe also rename it.